cmd/docker/docker.go

```diff
@@ -259,7 +260,18 @@ func runDocker(dockerCli *command.DockerCli) error {
 	// We've parsed global args already, so reset args to those
 	// which remain.
 	cmd.SetArgs(args)
-	return cmd.Execute()
+	err = cmd.Execute()
+	if err != nil {
+		return err
+	}
+
+	// If the command is being executed in an interactive terminal,
+	// run the plugin hooks (but don't throw an error if something misbehaves)
+	if term.IsTerminal(dockerCli.Out().FD()) {
```

Comment:
can use `Out.IsTerminal()` for this one;

```suggestion
	if dockerCli.Out().IsTerminal() {
```